### PR TITLE
[SPARK-33563][PYTHON][R][SQL] Expose inverse hyperbolic trig functions in PySpark and SparkR

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR adds the following functions (introduced in Scala API with SPARK-33061):

- `acosh`
- `asinh`
- `atanh`

to Python and R.

### Why are the changes needed?

Feature parity.

### Does this PR introduce _any_ user-facing change?

New functions.

### How was this patch tested?

New unit tests.

### CODE REVIEW

**Comment by zero323** [comment]
cc @rwpenney.

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -455,6 +455,19 @@ setMethod("acos",
             column(jc)
           })
 
+#' @details
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R:458]: @zero323, should we add them into `generic.R` too?

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -220,6 +220,19 @@ def acos(col):
     return _invoke_function_over_column("acos", col)
 
 
+def acosh(col):
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:223]: Let's also update `spark/python/docs/source/reference/pyspark.sql.rst`

**Comment by HyukjinKwon** [comment]
Looks fine otherwise.

File: R/pkg/R/functions.R (hunk 1 above)
Comment (zero323) [reply R/pkg/R/functions.R:458]: All trig functions are already generic, so we don't need that here.

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by zero323** [comment]
Thanks @HyukjinKwon @srowen!

**Pull request closed without merging**